apps/speed.c

```diff
@@ -613,17 +613,38 @@ static int EVP_Digest_loop(const char *mdname, ossl_unused int algindex, void *a
     unsigned char digest[EVP_MAX_MD_SIZE];
     int count;
     EVP_MD *md = NULL;
+    EVP_MD_CTX *ctx = NULL;
 
     if (!opt_md_silent(mdname, &md))
         return -1;
-    for (count = 0; COND(c[algindex][testnum]); count++) {
-        if (!EVP_Digest(buf, (size_t)lengths[testnum], digest, NULL, md,
-                        NULL)) {
+    if (EVP_MD_get_type(md) == NID_shake128
+        || EVP_MD_get_type(md) == NID_shake256) {
```

Comment:
Check for and XOF digest instead: `EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF`.

A side note, you should use `EVP_MD_is_a` rather than checking NIDs.  The latter fails for third party providers.

Reply:
changed to use `EVP_MD_get_flags(md) & EVP_MD_FLAG_XOF` and force pushed.